kernel/src/syscall.rs

```diff
@@ -616,6 +616,150 @@ fn return_scalar5(
+            }
+
+            (waiting_message, next_message)
+        };
+
+        // TODO: Have errors turn into calls to `ReceiveMessage`
+        let response = match result {
+            WaitingMessage::ScalarMessage(pid, tid) => {
+                let result = match scalar_type {
+                    1 => xous_kernel::Result::Scalar1(arg1),
+                    2 => xous_kernel::Result::Scalar2(arg1, arg2),
+                    _ => xous_kernel::Result::Scalar5(arg0, arg1, arg2, arg3, arg4),
+                };
+                MessageResponse { pid, tid, result }
+            }
+            WaitingMessage::ForgetMemory(_) => {
+                println!(
+                    "WARNING: Tried to wait on a scalar message that was actually forgetting memory"
+                );
+                return Err(xous_kernel::Error::ProcessNotFound);
+            }
+            WaitingMessage::BorrowedMemory(pid, tid, _server_addr, client_addr, len) => {
+                #[cfg(baremetal)]
+                let src_virt = _server_addr.get() as _;
+                #[cfg(not(baremetal))]
+                let src_virt = arg1 as _; // TODO: Check this is correct on hosted mode
+
+                // Return the memory to the calling process
+                ss.return_memory(src_virt, pid, tid, client_addr.get() as _, len.get())?;
+
+                MessageResponse {
+                    pid,
+                    tid,
+                    result: xous_kernel::Result::MemoryReturned(MemorySize::new(arg3), MemorySize::new(arg4)),
+                }
+            }
+            WaitingMessage::MovedMemory => {
+                println!(
+                    "WARNING: Tried to wait on a scalar message that was actually moved memory"
+                );
+                return Err(xous_kernel::Error::ProcessNotFound);
+            }
+            WaitingMessage::None => {
+                println!("WARNING: Tried to wait on a message that didn't exist");
+                return Err(xous_kernel::Error::ProcessNotFound);
+            }
+        };
+
+        // If there is a pending message, fetch it and schedule the thread to run
+        if let Some(msg) = next_message {
+            if cfg!(baremetal) {
+                // When the server is resumed, it will receive this as a return value.
+                ss.set_thread_result(
+                    server_pid,
+                    server_tid,
+                    xous_kernel::Result::MessageEnvelope(msg),
+                )?;
+                ss.ready_thread(response.pid, response.tid).unwrap();
+                ss.set_thread_result(response.pid, response.tid, response.result)?;
+                ss.activate_process_thread(server_tid, response.pid, response.tid, true)
+                    .map(|_| Ok(xous_kernel::Result::ResumeProcess))
+                    .unwrap_or(Err(xous_kernel::Error::ProcessNotFound))
+            } else {
+                println!("Rescheduling server because of full queue");
```

Comment:
same issue here, this is printing frequently.

Reply:
Removed this as well.